scale.lua

```diff
@@ -63,12 +66,26 @@ function Scale:GetScale()
     return scale.x, scale.y
 end
 
+function Scale:SetFromScale(x, y)
+    self.from.x = x
+    self.from.y = y
+end
+
+function Scale:GetFromScale()
+    local scale = self.from
+
+    return scale.x, scale.y
+end
+
 function Scale:OnUpdate(elapsed)
     local properties = self.group.properties
 
+	local fromx = self.from.x ~= nil and self.from.x or 1
+	local fromy = self.from.y ~= nil and self.from.y or 1	
```

Comment:
```suggestion
    local from_x = self.from.x
    local from_y = self.from.y
```

or if you prefer, use the references directly below, without assignment.

Reply:
if i didnt call SetFromScale() self.from.x\self.from.y will be nil. So this code just checks it
Mb we can put it in __Initialize() - self.from.x = 1 as default...